sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -2115,11 +2115,18 @@ def fill_matrix(*args):
               requires_partition_by=partitionings.Singleton(),
               proxy=proxy))
 
+  @frame_base.with_docs_from(pd.DataFrame)
   @frame_base.args_to_kwargs(pd.DataFrame)
   @frame_base.populate_defaults(pd.DataFrame)
   def corrwith(self, other, axis, drop, method):
-    if axis not in (0, 'index'):
-      raise NotImplementedError('corrwith(axis=%r)' % axis)
+    if axis in (1, 'columns'):
```

Comment:
What happens if I pass in 2?

Reply:
Usually what happens for bad arguments like this is that an error gets raised when we generate the proxy object, which is nice because it's an actual pandas error.

I went to verify that in this particular case, and we weren't actually raising an error because the `axis` argument wasn't passed through for proxy generation. I fixed this (and added a test) in 2a31d7c. So thank you for asking!